File: CausalInference/Udemy_Econometrics_and_Statistics_for_Business_in_R_n_Python/1. Difference-in-Differences/DID min wage.R
#pull the data
dataset <- read.csv("njmin3.csv")
summary(dataset)

#taking care of missing data
dataset$fte <- ifelse(is.na(dataset$fte),
                      mean(dataset$fte, na.rm = TRUE),
                      dataset$fte)
dataset$demp <- ifelse(is.na(dataset$demp),
                       mean(dataset$demp, na.rm = TRUE),
                      dataset$demp)
summary(dataset)

#create the first regression model
model1 <- lm(fte ~ NJ + POST_APRIL92 + NJ_POST_APRIL92,
             data = dataset)
summary(model1)

#create 2nd regression model
model2 <- lm(fte ~ NJ + POST_APRIL92 + NJ_POST_APRIL92 +
               bk + kfc + wendys,
             data = dataset)
summary(model2)

#Create third regression model
model3 <- lm(fte ~ NJ + POST_APRIL92 + NJ_POST_APRIL92 +
               bk + kfc + wendys +
               co_owned + centralj + southj,
             data = dataset)
summary(model3)

#visualize results
#install.packages("stargazer")
library(stargazer)
stargazer(model1, model2, model3,
          type = "text",
          title = "Impact of minimum wage on employment",
          no.space = TRUE,
          keep.stat = "n",
          digits = 2)

File: CausalInference/Udemy_Econometrics_and_Statistics_for_Business_in_R_n_Python/1. Difference-in-Differences/DiD min wage.py
#importing the libraries
import numpy as np
import pandas as pd

#pulling the data
dataset = pd.read_csv('njmin3.csv')

#check our data
dataset.describe()
pd.set_option('display.max_columns', None)

#check if there are more NAs
dataset.isnull().any()

#replacing NAs with averages
from sklearn.impute import SimpleImputer
missingvalues = SimpleImputer(missing_values = np.nan,
                              strategy = 'mean')
missingvalues = missingvalues.fit(dataset[['fte', 'demp']])
dataset[['fte', 'demp']] = missingvalues.transform(dataset[['fte', 'demp']])

#check if there are more NAs
dataset.isnull().any()

#isolating the X and Y variables
X = dataset.iloc[:, 0:3].values
Y = dataset.iloc[:, 3].values

#creating the first model
import statsmodels.api as sm
X = sm.add_constant(X)
model1 = sm.OLS(Y, X).fit()
model1.summary(yname = "FTE",
               xname = ("intercept", "New Jersey", "After April 1992",
                        "New Jersey and after April 1992"))

#isolating the X and Y variables part 2
X = dataset.loc[:, ['NJ', 'POST_APRIL92', 'NJ_POST_APRIL92',
                'bk', 'kfc', 'wendys']].values

#creating the second model
import statsmodels.api as sm
X = sm.add_constant(X)
model2 = sm.OLS(Y, X).fit()
model2.summary(yname = "FTE",
               xname = ("intercept", "New Jersey", "After April 1992",
                        "New Jersey and after April 1992",
                        "Burger King", "KFC", "Wendy's"))

#isolating the X and Y variables part 2
X = dataset.loc[:, ['NJ', 'POST_APRIL92', 'NJ_POST_APRIL92',
                'bk', 'kfc', 'wendys',
                'co_owned', 'centralj', 'southj']].values

#creating the third model
import statsmodels.api as sm
X = sm.add_constant(X)
model3 = sm.OLS(Y, X).fit()
model3.summary(yname = "FTE",
               xname = ("intercept", "New Jersey", "After April 1992",
                        "New Jersey and after April 1992",
                        "Burger King", "KFC", "Wendy's",
                        "Co-owned", "Central J", "South J"))










































































